docs/user-guide/02-conventional-features.md

```diff
@@ -581,3 +581,73 @@ Entry-point uniform parameters are semantically similar to global-scope shader p
 > #### Note ####
 > GLSL does not support entry-point `uniform` parameters; all shader parameters must be declared at the global scope.
 > Historically, HLSL has supported entry-point `uniform` parameters, but this feature was dropped by recent compilers.
+
+Mixed Shader Entry Points
+--------------------------
+
+Through the `[shader(...)]` syntax, users of slang can freely combine multiple entry points into the same file. This can be especially convenient for reuse between entry points which have a logical connection.
+
+For example, mixed entry points offer a convenient way for ray tracing applications to concisely define a complete pipeline in one source file, while also providing users with additional opportunities to improve type safety of 
+shared structure definitions:
+
+```hlsl
+struct Payload { float3 color; };
+
+[shader("raygeneration")]
+void rayGenerationProgram() {
+    Payload payload;
+    TraceRay(/*...*/, payload);
+    /* ... */ 
+}
+
+[shader("closesthit")]
+void closestHitProgram(out Payload payload) { 
+    payload.color = {1.0};
+}
+
+[shader("miss")]
+void missProgram(out Payload payload) { 
+    payload.color = {1.0};
+}
+```
+
+> #### Note ####
+> GLSL does not support multiple entry-points; however, SPIR-V does. Vulkan users wanting to take advantage of Slang mixed entry points must pass `-fvk-use-entrypoint-name` and `-emit-spirv-directly` as compiler arguments.
+
+### Mixed Entry-Point Uniform Parameters
+
+Like with the previous `vertexMain` example, mixed entry point setups also support _entry-point uniform parameters_.
+
+However, because of certain systematic differences between entry point types, a uniform being _global_ or _local_ will have very important consequences on the underlying layout and behavior.
+
+For most all entry point types, D3D12 and Vulkan use one common root signature to define both global and local uniform parameters. For these cases, Slang maps uniforms to this common root signature. 
+
+However, for ray tracing entry points, these parameters map to either _global_ root signatures or to _local_ root signatures, with the latter being stored in the shader binding table.
```

Comment:
A parameter may not always be visible in a root signature. An ordinary data field will typically be allocated in the global default constant buffer, which does not correspond directly to any entries in the root signature.

This should be reworded to something like:

> The scope where a uniform parameter is defined has a consequence on the parameter binding in the target code. For ray tracing shaders, uniform parameters defined in the global scope will determine the shape of global root signature / pipeline layout, while uniform parameters defined as entry point function parameters will be allocated in Shader Binding Tables. On D3D12, uniform entry-point parameters will affect the local root signature of the entry-point.

Reply:
This paragraph can be moved to the "EntryPoint Shader Parameters" section above.

Reply:
> global default constant buffer, which does not correspond directly to any entries in the root signature

I did not know this, but that makes sense. I'm less familiar with D3D12, but in Vulkan, the same would be true for push constants. The above revision seems good to me.